### DISCUSSION TITLE
Questions around BinningProcess() and how it handles missing values and monotonic trend

### DISCUSSION THREAD

**Discussion by priyankamishra31** [guillermo-navas-palencia/optbinning#332] [General]
**Missing Values**
Can you please tell me how BinningProcess() handles missing value in dataset ? Does the process assign a random value to them or have them as a separate bin ?

**Monotonic Trend**
Do all the bins returned by the BinningProcess() follow monotonic trend by default?

Thanks for your help with this

**Comment by bmreiniger** [comment]
Missing values get their own bin, see e.g. [this subsection of the tutorial](http://gnpalencia.org/optbinning/tutorials/tutorial_binary.html#Missing-data-and-special-codes).

Monotonic trends get detected. See [this subsection of the tutorial](http://gnpalencia.org/optbinning/tutorials/tutorial_binary.html#Event-rate-/-WoE-monotonicity), and the docstring of `monotonic_trend` of the `OptimalBinning` and related objects ([docs for OptimalBinning](http://gnpalencia.org/optbinning/binning_binary.html)), which [defaults to](https://github.com/guillermo-navas-palencia/optbinning/blob/aebe16b4d5f32662d355cac1cbf01056f2246eb2/optbinning/binning/binning_process.py#L91) `"auto"` when using the `BinningProcess`.
